1-producer-consumer/main.go

```diff
@@ -13,19 +13,20 @@ import (
 	"time"
 )
 
-func producer(stream Stream) (tweets []*Tweet) {
+func producer(stream Stream, ch chan<- *Tweet) {
```

Comment:
Використовуючи паттерн "генератор", код виходить красивішим:

```go
func producer(stream Stream) chan *Tweet {
	tweets := make(chan *Tweet)
	go func() {
		for {
			tweet, err := stream.Next()
			if err == ErrEOF {
				close(tweets)
				return
			}

			tweets <- tweet
		}
	}()
	return tweets
}

// ...

func main() {
	start := time.Now()
	stream := GetMockStream()
	// Producer
	tweets := producer(stream)

	// Consumer
	consumer(tweets)

	fmt.Printf("Process took %s\n", time.Since(start))
}
```

Reply:
https://go.dev/talks/2012/concurrency.slide#25